src/User/Types.hs

```diff
@@ -28,13 +28,13 @@ instance PathPiece Currency where
   toPathPiece = T.pack . printf "%.2f" . unCurrency
 
 data User = User
-  { userKey       :: Int
-  , userName      :: String
-  , userEmail     :: String
-  , userBirthDate :: Day
-  , userPassword  :: String
-  , userGender    :: Gender
-  , userIncome    :: Currency}
+  { userKey       :: !Int
+  , userName      :: !String
+  , userEmail     :: !String
+  , userBirthDate :: !Day
+  , userPassword  :: !String
+  , userGender    :: !Gender
+  , userIncome    :: !Currency}
```

Comment:
:+1: check https://wiki.haskell.org/Performance/Data_types#Unpacking_strict_fields

maybe you want to add `{-# OPTIONS_GHC -funbox-strict-fields #-}` at the top of your file

Reply:
Sure thing, gonna add that pragma right now.